### ISSUE TITLE
Archaea?

### ISSUE THREAD

**Issue by berthubert** [AllTheBacteria/AllTheBacteria#28]
Firstly, I am in AWE of your work, and I hope to marry it with https://skewdb.org/ at my earliest convenience. This is database of GC skews of bacterial and archaeal chromosomes. 

I downloaded https://ftp.ebi.ac.uk/pub/databases/AllTheBacteria/Releases/0.2/metadata/species_calls.tsv.gz to find my favorite archaeum, and also perhaps the most famous one, https://en.wikipedia.org/wiki/Methanobrevibacter_smithii and it appears not to be in there. I did spot one other methanobrevibacter in there though.

It is entirely possible your project has decided to focus on eubacteria, but it would be good to know. 

Or perhaps this is an issue with the species caller not including archaea? Or perhaps I am misunderstanding the metadata?

In any case, anything I could do to help, please do let me know, especially if this involves programming or data processing.

Thanks!

**Comment by martinghunt** [comment]
Hi there,

Archaea wasn't included. We took everything underneath taxonomy ID 2, which is bacteria.

The species calls were made with Sylph, which uses GTDB, which I think incudes archaea. The call of methanobrevibacter is probably contamination or some error with the ENA metadata saying bacteria, but the actual sequencing reads having something else. Species calling is a whole other topic (we're looking into handle it better) and there will be some errors in the calls.

I did just have a quick look at archaea - getting all sequencing runs from the ENA under taxonomy ID 2157. There's only 13,607 runs. For AllTheBacteria, we then need Illumina, paired, genomic and WGS sequencing. This leaves just 816 sequencing runs. Not sure how much use those are, but I can easily put them through the assembly pipeline.

We want to welcome any new analysis, and are in the process of writing documentation for how anyone can contribute to the project. But of course anyone can download the data and do what they like with it.

**Comment by berthubert** [comment]
Hi Martin, thanks for the swift reply! I've now found a few more M smithii in sylph.tsv.gz but this appears to mostly be contaminants. In the NCBI ftp.ncbi.nlm.nih.gov/genomes/refseq/archaea/assembly_summary.txt database there are around 600 complete Archaeal genomes, so 816 from the ENA is not a huge difference. 

Archaea suffer a bit from lack of visibility, but they are of exceptional importance to understand where eukaryotes came from, so I'd very much recommend including them in AllTheBacteria. 

With regards to helping people contribute to the project, I'm writing a small file of things I've discovered, and I'll share that with you when I'm done. It is likely at most 90% correct but it might be of use already.

Good luck with your important work!

**Comment by martinghunt** [comment]
Righto, I'll add in those 816. They can be included in our next release.

Thanks, the more analysis the better! I'll leave this issue open, please reply to it when you're done

**Comment by ctb** [comment]
This whole conversation made me curious so I thought I'd dive in with a little [sourmash](https://sourmash.readthedocs.io/). Treating the sample as a potential mixture, I got the following results when analyzing it against GTDB rs214 - results truncated for ease of viewing.

tl;dr it's probably got an unknown d__Bacteria;p__Pseudomonadota in it? and/or a mixture of other crud. Not sure. More analysis needed.

At the family level:
```
sample name    proportion   cANI   lineage
-----------    ----------   ----   -------
SAMEA104402294    34.2%     -      unclassified
SAMEA104402294     5.1%     86.1%  d__Bacteria;p__Pseudomonadota;c__Gammaproteob
acteria;o__Steroidobacterales;f__Steroidobacteraceae
SAMEA104402294     4.5%     85.6%  d__Bacteria;p__Pseudomonadota;c__Alphaproteob
acteria;o__Rhizobiales;f__Xanthobacteraceae
SAMEA104402294     4.1%     85.4%  d__Bacteria;p__Actinomycetota;c__Actinomycetia;o__Streptomycetales;f__Streptomycetaceae
SAMEA104402294     3.0%     83.3%  d__Bacteria;p__Pseudomonadota;c__Alphaproteobacteria;o__Rhizobiales;f__Rhizobiaceae
SAMEA104402294     3.0%     84.5%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae
SAMEA104402294     2.3%     84.0%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Xanthomonadales;f__Xanthomonadaceae
SAMEA104402294     2.3%     83.0%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Burkholderiales;f__SG8-39
SAMEA104402294     2.3%     84.0%  d__Bacteria;p__Pseudomonadota;c__Alphaproteobacteria;o__Sphingomonadales;f__Sphingomonadaceae
SAMEA104402294     2.1%     83.7%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae_B
```

At the species level:
```
sample name    proportion   cANI   lineage
-----------    ----------   ----   -------
SAMEA104402294    34.2%     -      unclassified
SAMEA104402294     3.1%     83.7%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Steroidobacterales;f__Steroidobacteraceae;g__Steroidobacter_A;s__Steroidobacter_A gossypii
SAMEA104402294     2.3%     83.0%  d__Bacteria;p__Pseudomonadota;c__Alphaproteobacteria;o__Rhizobiales;f__Xanthobacteraceae;g__Bradyrhizobium;s__Bradyrhizobium canariense
SAMEA104402294     1.8%     82.2%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Burkholderiales;f__SG8-39;g__SCGC-AG-212-J23;s__SCGC-AG-212-J23 sp005881595
SAMEA104402294     1.8%     79.9%  d__Bacteria;p__Bacillota;c__Bacilli;o__Lactobacillales;f__Enterococcaceae;g__Enterococcus_B;s__Enterococcus_B faecium
SAMEA104402294     1.2%     79.9%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia;s__Escherichia coli
SAMEA104402294     1.2%     81.2%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Xanthomonadales;f__Xanthomonadaceae;g__Lysobacter;s__Lysobacter enzymogenes_B
SAMEA104402294     1.2%     79.9%  d__Bacteria;p__Myxococcota;c__Polyangia;o__Haliangiales;f__Haliangiaceae;g__JABFXX01;s__JABFXX01 sp016720595
SAMEA104402294     1.2%     79.1%  d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Xanthomonadales;f__UBA2363;g__Pseudofulvimonas;s__Pseudofulvimonas sp023228845
...
```

## What I ran

You'll need `gtdb-rs214-k21.zip` and `gtdb-rs214.lineages.csv.gz` from https://sourmash.readthedocs.io/en/latest/databases.html.

```
mamba create -n smash -y sourmash_plugin_branchwater
mamba activate smash

# sketch genome (already done by @ccbaumler as part of https://github.com/AllTheBacteria/AllTheBacteria/issues/23) and put in `signatures/7a15c805a0fb99ef39f14b67b54441d7.sig.gz`
# e.g. run:
# sourmash sketch dna -p k=21,k=31,k=51 GENOME.FASTA -o signatures/7a15c805a0fb99ef39f14b67b54441d7.sig.gz

# analyze sketch as metagenome => decompose into mixture of genomes
sourmash scripts fastgather signatures/7a15c805a0fb99ef39f14b67b54441d7.sig.gz gtdb-rs214-k21.zip --threshold-bp=0 -o test2.csv -k 21

# classify taxonomy at family level
sourmash tax metagenome -g test2.csv -t gtdb-rs214.lineages.csv.gz  -F human -r family

# classify taxonomy at species level
 sourmash tax metagenome -g test2.csv -t gtdb-rs214.lineages.csv.gz  -F human -r family
```

**Comment by shenwei356** [comment]
@ctb That sample is quite complex. 

It's fragmented.

```
$ seqkit stats SAMEA104402294.fa.gz
file                  format  type  num_seqs  sum_len  min_len  avg_len  max_len
SAMEA104402294.fa.gz  FASTA   DNA      1,980  700,786      200    353.9    1,708
```
And only 21.1616% (419/1980) contigs can be aligned to GTDB r214 representative genomes.

And the matched species of different contigs  are very different

```
query                        qlen   hits   sgenome           sseqid                 qcovGnm   hsp   qcovHSP   alenHSP   pident    gaps   qstart   qend   sstart    send      sstr   slen       species                                                  
--------------------------   ----   ----   ---------------   --------------------   -------   ---   -------   -------   -------   ----   ------   ----   -------   -------   ----   --------   ---------------------------------------------------------
SAMEA104402294.contig00004   1598   37     GCF_000972865.1   NZ_CP011144.1          98.310    1     98.310    1573      91.100    7      1        1571   1240032   1241599   +      3891720    Pseudoxanthomonas suwonensis                             
SAMEA104402294.contig00004   1598   37     GCA_024707405.1   JANPZU010000006.1      98.874    1     98.874    1582      90.518    7      1        1580   120529    122105    -      184814     Luteimonas sp.                                           
SAMEA104402294.contig00004   1598   37     GCA_023419315.1   JAEZYE010000018.1      98.310    1     98.310    1573      90.083    7      1        1571   29766     31333     -      64393      Pseudomonadota bacterium                                 
SAMEA104402294.contig00042   893    22     GCA_004297335.1   SCUD01000044.1         94.289    1     94.289    842       90.261    0      1        842    6590      7431      +      15499      Gammaproteobacteria bacterium                            
SAMEA104402294.contig00049   842    39     GCA_019634895.1   JAHCAM010000029.1      98.219    1     98.219    833       91.116    19     1        827    27986     28805     -      31158      Steroidobacteraceae bacterium                            
SAMEA104402294.contig00049   842    39     GCA_016874295.1   VGUH01000017.1         98.219    1     98.219    836       90.789    13     1        827    58482     59313     -      63986      Gammaproteobacteria bacterium                            
SAMEA104402294.contig00049   842    39     GCA_016882265.1   VGVB01000048.1         60.451    1     60.451    509       94.695    0      1        509    1         509       -      20535      Gammaproteobacteria bacterium                            
SAMEA104402294.contig00051   832    100    GCF_010120715.1   NZ_CP047591.1          97.716    1     97.716    813       95.080    0      1        813    1737162   1737974   +      3512404    Aminipila terrae                                         
SAMEA104402294.contig00051   832    100    GCF_019203185.1   NZ_CP078093.1          95.673    1     95.673    797       90.966    1      1        796    1007467   1008263   +      2756133    Crassaminicella indica                                   
SAMEA104402294.contig00051   832    100    GCF_019884785.1   NZ_CP082326.1          95.673    1     95.673    797       90.339    1      1        796    4422982   4423778   +      4589101    Crassaminicella profunda                                 
SAMEA104402294.contig00051   832    100    GCF_008152325.1   NZ_CP042243.1          95.673    1     95.433    795       90.566    1      1        794    1813267   1814061   -      3046753    Crassaminicella thermophila                              
SAMEA104402294.contig00065   787    93     GCF_003173755.1   NZ_CP029553.1          80.432    1     80.432    633       91.311    2      108      740    2179438   2180068   -      6162702    Methylobacterium terrae                                  
SAMEA104402294.contig00065   787    93     GCF_003058325.1   NZ_CP028843.1          80.432    1     80.432    633       91.311    2      108      740    15947     16577     +      6553813    Methylobacterium currus                                  
SAMEA104402294.contig00065   787    93     GCF_019216885.1   NZ_JAAQRC020000002.1   80.432    1     80.432    633       91.153    2      108      740    330171    330801    +      742825     Methylobacterium sp. SyP6R                               
SAMEA104402294.contig00065   787    93     GCF_003173775.1   NZ_CP029552.1          80.432    1     80.432    633       90.837    2      108      740    18624     19254     +      6542583    Methylobacterium sp. 17Sr1-1                             
SAMEA104402294.contig00065   787    93     GCF_003119115.1   NZ_CP022363.1          80.432    1     80.432    633       90.047    2      108      740    429310    429940    +      836998     Azospirillum sp. TSH58                                   
SAMEA104402294.contig00073   754    85     GCA_019683695.1   JADGHX010000146.1      93.369    1     93.369    704       96.449    0      10       713    8738      9441      -      12997      Steroidobacteraceae bacterium                            
SAMEA104402294.contig00073   754    85     GCA_019634915.1   JAHCAN010000006.1      88.462    1     88.462    667       94.903    0      11       677    95960     96626     +      118698     Steroidobacteraceae bacterium                            
SAMEA104402294.contig00073   754    85     GCA_016720075.1   JADJXW010000006.1      88.594    1     88.594    668       93.862    0      10       677    11554     12221     -      256526     Pseudomonadota bacterium                                 
SAMEA104402294.contig00073   754    85     GCA_016707425.1   JADJHU010000015.1      87.931    1     87.931    663       94.118    0      11       673    886300    886962    -      1028894    Gammaproteobacteria bacterium                            
SAMEA104402294.contig00073   754    85     GCA_945860725.1   CAMAQZ010000005.1      87.401    1     87.401    659       93.171    0      10       668    45126     45784     -      65409      Klebsiella pneumoniae                                    
SAMEA104402294.contig00111   657    75     GCF_020535565.1   NZ_CP084758.1          91.172    1     91.172    599       92.821    0      1        599    312127    312725    +      5159983    Shinella sp. XGS7                                        
SAMEA104402294.contig00111   657    75     GCF_014117425.1   NZ_CP059725.1          84.018    1     84.018    552       93.116    0      1        552    2820247   2820798   +      3368120    Bartonella sp. HY038                                     
SAMEA104402294.contig00111   657    75     GCF_000278115.1   NZ_JH725117.1          84.018    1     84.018    552       93.116    0      1        552    6020      6571      +      246143     Bartonella sp. DB5-6                                     
SAMEA104402294.contig00111   657    75     GCF_000518165.1   NZ_KI912393.1          84.018    1     84.018    552       93.116    0      1        552    382       933       -      75830      Bartonella elizabethae                                   
SAMEA104402294.contig00111   657    75     GCF_003173735.1   NZ_CP029551.1          82.801    1     82.801    544       91.360    0      8        551    374444    374987    +      5539695    Methylobacterium radiodurans
```
[SAMEA104402294.fa.gz.align.tsv.add_species.filtered.txt](https://github.com/user-attachments/files/16100810/SAMEA104402294.fa.gz.align.tsv.add_species.filtered.txt)

**Comment by iqbal-lab** [comment]
original issue is closed
